posts/test_post_uploading.py

```diff
@@ -0,0 +1,124 @@
+"""Unit tests for posts uploading."""
+
+# Authors: mukobi
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import mongomock
+import app
+
+VALID_POST_FULL = {
+    "event_id": "abc123",
+    "author_id": "kurt_vonnegut",
+    "text": "This is a very valid post.",
+    "files": [
+        "Pretend I am a long base-64 encoded file",
+        "Another pretend encoded file"
+    ]}
+VALID_POST_TEXT_NO_FILES = {
+    "event_id": "abc123",
+    "author_id": "ray_bradbury",
+    "text": "This post has no files but is still valid.",
+    "files": []}
+VALID_POST_FILES_NO_TEXT = {
+    "event_id": "abc123",
+    "author_id": "douglas_adams",
+    "text": "",
+    "files": [
+        "This post has no text but is valid because it has at least one file."
+    ]}
+INVALID_POST_NO_TEXT_NOR_FILES = {
+    "event_id": "abc123",
+    "author_id": "jrr_tolkien",
+    "text": "",
+    "files": []}
+INVALID_POST_NOT_ENOUGH_ATTRS = {
+    "where_did_all_the_attributes_go?": "I don't know"}
+INVALD_POST_TOO_MANY_ATTRS = {
+    "event_id": "abc123",
+    "author_id": "grr_martin",
+    "text": "I have a valid description but too many fields.",
+    "files": [],
+    "woah_where_did_I_come_from?": "malicious data"}
+
+
+class TestPostUploading(unittest.TestCase):
+    """Test post uploading of users service."""
+
+    def assert_posts_are_equal(self, first, second):
+        """Asserts that the 2 posts have equal data.
+
+        Only looks at required post attributes
+        (event_id, author_id, text, files).
+        """
+        self.assertEqual(first["event_id"], second["event_id"])
+        self.assertEqual(first["author_id"], second["author_id"])
+        self.assertEqual(first["text"], second["text"])
+        self.assertEqual(first["files"], second["files"])
+
+    def setUp(self):
+        """Set up mock DB for testing"""
+        self.mock_collection = mongomock.MongoClient().db.collection
+
+    def test_full_upload(self):
```

Comment:
You seem to have two cases here:

1. Post a certain payload, expect that object to exist.
2. Post a certain payload, expect no object to exist.

This would be a good use case for table-driven testing, but it appears that table-driven tests are harder to write in python than in Go:

http://tomlinford.com/post/table-driven-tests/ is one reference; these are sometimes also called "parameterized tests".

Reply:
So under table testing all these tests would be in a single function, correct? I'll look into implementing this later.